project/make/package-rpm

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+
+DEST=$1
+
+PKGVERSION="${VERSION//-/'~'}"
+# if we have a "-dev" suffix or have change in Git, let's make this package version more complex so it works better
+if [[ "$VERSION" == *-dev ]] || [ -n "$(git status --porcelain)" ]; then
+	GIT_UNIX="$(git log -1 --pretty='%at')"
+	GIT_DATE="$(date --date "@$GIT_UNIX" +'%Y%m%d.%H%M%S')"
+	GIT_COMMIT="$(git log -1 --pretty='%h')"
+	GIT_VERSION="git${GIT_DATE}.0.${GIT_COMMIT}"
+	# GIT_VERSION is now something like 'git20150128.112847.0.17e840a'
+	PKGVERSION="$PKGVERSION~$GIT_VERSION"
+fi
+
+# probably shouldn't be using dpkg here...
+PACKAGE_ARCHITECTURE="$(dpkg-architecture -qDEB_HOST_ARCH)"
```

Comment:
oh dear...

Something like ` rpm --eval "%{_arch}"` on rpm hosts. Though, since this is an parameter passed to FPM, I'm not sure whether fpm is expecting debian `amd64` or rpm `x86_64` style.

Also, since this code should be agnostic to the host running it (don't call dpkg from rpm systems, and vice versa), I recommend going a cleaner route of a `uname -m` and case statement. Not terribly unlike what we do for exporting arch variables on slackware http://slackware.osuosl.org/slackware64-current/source/a/bash/bash.SlackBuild